### DISCUSSION TITLE
Achievement categories

### DISCUSSION THREAD

**Discussion by Jamiras** [RetroAchievements/RAWeb#744] [Ideas]
This is related to #718, but I decided to keep it as a separate issue as it's an alternate solution. It also addresses #687.

The concept is to categorize each achievement.
* Progress - achievement is required to complete the game
* Optional - achievement is not required to complete the game
* Easter Egg - achievement is associated with finding something hidden
* Challenge - achievement requires additional effort, but should be something an average user could complete with practice
* Expert - achievement requires an abnormal amount of effort

This would also be a good place to flag things as Missable (cannot be earned without restarting the game), but that should probably be a separate modifier: i.e. Missable Optional, Missable Easter Egg, Missable Challenge.

Players would be awarded for "beating the game" by earning all Progress achievements. For each additional category the player completes, they could earn another star for their mastery frame. Or, rather than having stars, have indicators for each category, and a single star indicating mastery of all categories (maybe have the star just be the non-Expert ones):

![image](https://user-images.githubusercontent.com/32680403/45705204-349afa00-bb36-11e8-911d-0a541372a50a.png)

Those icons (or something similar) could also be applied to the individual achievements showing which category they're associated to.

`~Bonus~` achievements would be moved into the set as Expert achievements. `~Multi~` achievements could be treated as a separate category, but comments in #718 suggest they deserve something different. Comments also suggest that some bonus achievements (like glitching) should be left out of the Expert group. These should still be available to the player without having to patch their ROM; preferably without having to specifically enable them.

**Comment by rzumer** [comment]
I believe that my comments in #56 still stand. The distinction between "optional", "challenge" and "expert" also seems to be arbitrary. The line is not so easy to draw.

Giving so much information about how to unlock things is not only a burden for the developer (and subject to error), but defeats part of the purpose of the achievement system (at least for [Bartle's explorers](http://mud.co.uk/richard/hcds.htm)). I believe that achievement systems on common platforms are kept simple at least in part for those reasons.

**Comment by meleu** [comment]
I agree with what @rzumer said above.
I predict many heated (and unnecessary) discussions between devs to define what should be "optional", "challenge" or "expert".

However, regarding progress and the "beaten" trophy (#56) I would like to see it implemented at some point.

**Comment by SyrianBallaS** [comment]
Yeah this stuff does seem arbitrary and will lead to many people arguing what's the "right way" when there really isn't one from a requirements stand point. Requirements must be solid and unambigious (no what-ifs).

I agree that beaten and progress trophies should be made eventually but should be optional still.

**Comment by Jamiras** [comment]
Everything is already arbitrary. 20 pts for completing the first level? Sounds good. No damage achievements in a beat-em-up? Put it in core. Grind to max level in an RPG? Gotta have that for mastery.

This is just taking a concept similar to the one in http://retroachievements.org/game/2032 and expanding it such that it could be applied to all games.

I agree that the "beaten" (i.e. progress) bucket is probably the most important one, so people who don't want to do the optional/challenge stuff can still get a sticker on the profile, but why not label everything else too?

**Comment by rzumer** [comment]
You are correct, there are many arbitrary and contentious choices embedded in the design already. That is not a good thing, and we should not be adding more without a very good reason.

Some amount of creativity is necessary in set composition. For the record, I am in favor of removing the static point system.

On September 19, 2018 5:11:44 PM EDT, Jamiras <notifications@github.com> wrote:
>Everything is already arbitrary. 20 pts for completing the first level?
>Sounds good. No damage achievements in a beat-em-up? Put it in core.
>Grind to max level in an RPG? Gotta have that for mastery.
>
>This is just taking a concept similar to the one in
>http://retroachievements.org/game/2032 and expanding it such that it
>could be applied to all games.
>
>I agree that the "beaten" (i.e. progress) bucket is probably the most
>important one, so people who don't want to do the optional/challenge
>stuff can still get a sticker on the profile, but why not label
>everything else too?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/RetroAchievements/RAWeb/issues/221#issuecomment-422960087

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by ScottFromDerby** [comment]
Edit: moved comment to https://github.com/RetroAchievements/RAWeb/issues/225

**Comment by kdecks** [comment]
I love this

**Comment by meleu** [comment]
@ScottFromDerby it seems unrelated to the topic being discussed here. Could you please open a new issue with this specific feature request?

**Comment by kdecks** [comment]
@ScottFromDerby removed it but never created a new issue.... It was a great idea. (On the game page, showing that an achievement has a guide video by displaying a video icon by the achievement that has a video)

**Comment by ScottFromDerby** [comment]
@kdecks  moved it to here https://github.com/RetroAchievements/RAWeb/issues/225

**Comment by Prism7** [comment]
Recently, I've been thinking more about the overall concept of separating achievements into a standard set of groups, and I figured that I would share my thoughts. Hopefully, I'm posting this in the right issue. I know that I'm borrowing parts from several of them.

To start, I would separate every achievement that is directly related to in-game progress and official side quests. I would label this category the "Core" group. Much like with the current system, earning everything in this "Core" group would earn you the standard badge. In addition to this, I think that it might be beneficial if developers could flag a single achievement (possibly several) within this "Core" group as an indicator for having beaten/completed the main game. This would be optional, and obviously, it wouldn't work for every game; however, for most games, if a player earns this marked achievement(s), then the player would be awarded a black and white version of the game icon (using the locked filter).

I like the idea of additional groups, but it might be necessary to limit them to an agreed upon number, maybe 4 or 5. I also agree that the badge stickers would be the perfect default rewards for completing these extra groups, but someone will need to find a way to have the badges update when the number of achievements in a category changes (e.g. the sticker is removed from all badges after a developer adds new achievements to the category).

In an attempt to create some sort of standardization, I think the first two additional groups for every set should be "Standard Challenges" and "Extreme Challenges". Many of the achievements in "Standard Challenges" would be ones that currently exist in the core set. Some examples being most "deathless" achievements, speedruns (within reason), most level-related achievements in RPG's, most boss challenges, most minigame challenges, etc. The "Extreme Challenges" category would be for the various achievements that often wind up in bonus sets, the ones that are designed for the most hardcore players who have played through the game more times than they can count and are looking for a new reason to continue playing. More specifically, it could house combination challenges such as "complete ? with/without doing X, Y, and Z", extreme low level challenges in RPG's, achievements requiring perfection for an extended period of time, achievements relying on random chance when the odds are very low, anything requiring excessive grinding, etc.

The final 2-3? groups would be dependent upon the game. Some examples might be "Cooperative/Multi-Player", "Glitches", "Joke/Meme", "Region/Version Specific", etc. It might also be useful to have the final group for most games be "Miscellaneous"; this would allow developers to code a random achievement here and there that either doesn't belong to a recognized category or is an outlier (like when a game only has a few known glitches or there's only enough region-specific content for a couple of achievements).

Last but not least, to further incentivize players to try to complete as many groups as possible (and also separate Hardcore users), I would apply different borders to the badges. Bronze, Silver, and Gold would probably be the most obvious choices. The bronze border could be awarded for earning all "Core" achievements in Hardcore, the silver for earning all achievements in both the "Core" and the "Standard Challenges" groups in Hardcore, and the gold for earning all achievements in every group in Hardcore. On a final note, I think it's best if the stickers were only awarded to Hardcore users. Softcore players would only earn the badges, without the borders or stickers.

**Comment by Blazekickn** [comment]
Since this is being brought back from the dead

![image](https://user-images.githubusercontent.com/38176599/67737594-26c6d200-f9e2-11e9-9fb5-f777691e18b6.png)

more what i had in mind. I'm fine with awarding badges on beating and completing a game just make them less appealing in comparison (aka no border for you until you actually do all the challenges). The piss colored border might not be the prettiest thing ever but (aside from it being easy to edit) it's what we're used to. When you see a yellow border, you know that they mastered the game and earned all of the challenges in the set. There isn't much of a need to overcomplicate it with percentages and whatnot. Witholding the next rank of badge until you earn everything in a rank can be a good thing in encouraging people to go for everything instead of just giving up and walking away with a silver.

You can add stickers or whatever and maybe add an extra tier for completing all the bonus stuff that isn't Mr. Perfect as long as it's not obnoxious to look at. Simplicity is key here.

**Comment by kdecks** [comment]
Maybe mastery could be rainbow. I think it's important that mastery implies 100% and is the top tier badge, and getting all the sticker/star groups is separate. This is for completionist's sake. Anything above 100%, bonus groups although giving recognition shouldn't be implied as "more complete". And in some cases it won't be good for all players to finish every bonus group, for example, multiplayer achievements, or achievements that are specific to a regional hash. 

--- 

I'll also link the latest write up of how I see this all fitting together:
http://retroachievements.org/viewtopic.php?t=9136&c=48754

I'd like to disclaim that this isn't "my" idea, but is more that I've gathered many of the ideas put forward by the community here into one place, and I'm sure that it as a concept still has plenty of room for improvement.

**Comment by Tiger15cub** [comment]
how do you even add achevements to the one you self host?

**Comment by wescopeland** [comment]
https://github.com/RetroAchievements/RAWeb/pull/1714 adds support for achievement categories.
